File: yangxj96-serve/yangxj96-serve-auth/src/main/java/io/github/yangxj96/server/auth/service/RoleService.java
/*****************************
 * Copyright (c) 2021 - 2023
 * author:yangxj96
 * email :yangxj96@gmail.com
 * date  :2023-01-07 00:08:39
 * Copyright (c) 2021 - 2023
 ****************************/

package io.github.yangxj96.server.auth.service;

import io.github.yangxj96.bean.user.Role;
import io.github.yangxj96.common.base.BasicService;

/**
 * 角色service层
 *
 * @author yangxj96
 * @version 1.0
 * @date 2023-01-07 00:14
 */
public interface RoleService extends BasicService<Role> {

    /**
     * 关联权限
     *
     * @param role      角色id
     * @param authority 权限id
     * @return 是否关联成功
     */
    boolean relevance(Long role, Long authority);
}

File: yangxj96-serve/yangxj96-serve-auth/src/main/java/io/github/yangxj96/server/auth/service/RoleService.kt
package io.github.yangxj96.server.auth.service;

import cn.hutool.core.lang.tree.Tree;
import io.github.yangxj96.bean.user.Role;
import io.github.yangxj96.common.base.BasicService;

import java.util.List;

/**
 * 角色service层
 */
public interface RoleService extends BasicService<Role> {

    /**
     * 关联权限
     *
     * @param role      角色id
     * @param authority 权限id
     * @return 是否关联成功
     */
    boolean relevance(Long role, Long authority);

    /**
     * 返回树格式的角色列表
     *
     * @return 树结构的角色列表
     */
    List<Tree<String>> tree();
}
